pkg/genericcli/context.go

```
@@ -0,0 +1,682 @@
+		},
+		ListCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Args = cobra.ExactArgs(0)
+		},
+		CreateCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().String(KeyName, "", "set the name of the context")
+			cmd.Flags().String(KeyAPIURL, "", "set the api-url for this context")
+			cmd.Flags().String(KeyAPIToken, "", "set the api-token for this context")
+			cmd.Flags().String(KeyDefaultProject, "", "set a default project to operate on")
+			cmd.Flags().Duration(KeyTimeout, 0, "set a default request timeout")
+			cmd.Flags().Bool(KeyActivate, false, "immediately switches to the new context")
+			cmd.Flags().String(KeyProvider, "", "set the login provider for this context")
+
+			Must(cmd.MarkFlagRequired(KeyName))
+			Must(cmd.MarkFlagRequired(KeyAPIToken))
+
+			cmd.Args = cobra.ExactArgs(0)
+		},
+		UpdateCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().String(KeyAPIURL, "", "set the api-url for this context")
+			cmd.Flags().String(KeyAPIToken, "", "set the api-token for this context")
+			cmd.Flags().String(KeyDefaultProject, "", "set a default project to operate on")
+			cmd.Flags().Duration(KeyTimeout, 0, "set a default request timeout")
+			cmd.Flags().Bool(KeyActivate, false, "immediately switches to the new context")
+			cmd.Flags().String(KeyProvider, "", "set the login provider for this context")
+
+			Must(cmd.RegisterFlagCompletionFunc(KeyDefaultProject, c.ProjectListCompletion))
+
+			cmd.ValidArgsFunction = c.ContextListCompletion
+
+			cmd.Args = cobra.ExactArgs(1)
+		},
+		DeleteCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.ValidArgsFunction = c.ContextListCompletion
+
+			cmd.Args = cobra.ExactArgs(1)
+		},
+		CreateRequestFromCLI: func() (*Context, error) {
+			return &Context{}, nil // Placeholder to trigger cmdline read (not file read)
+		},
+		UpdateRequestFromCLI: func(args []string) (*contextUpdateRequest, error) {
+			name, err := GetExactlyOneArg(args)
+			if err != nil {
+				return nil, fmt.Errorf("no context name given")
```

Swallow of original error, `err` should be returned.